### PR TITLE
build bundle with assemble

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Update build bundle to delegate the call to build bundle to flutter assemble. Then copy the required assets out of the cache directory and into the build/* directory, or where told.

Problem: depfile generated by dart rule will have the wrong paths. I don't really think we should hack around this, and instead work on shrinking flutter gradle to a single assemble invocation. The tool can then output inputs/outputs in a depfile format

### CODE REVIEW

**Comment by jonahwilliams** [comment]
This needs to support codegen too....

**Comment by jonahwilliams** [comment]
Updated to use the new OUTPUT_DIR pattern, and to produce a correct depfile for gradle (it was excluding all of the flutter assets).

@blasten, were the flutter assets being included as a dependency in some other way?

**Review by jmagman — APPROVED** [review]
LGTM

**Comment by jmagman** [comment]
```
Unused import: 'build_system/targets/assets.dart' • packages/flutter_tools/lib/src/bundle.dart:16:8 • unused_import
```

**Comment by jonahwilliams** [comment]
This change requires https://github.com/flutter/flutter/pull/39654 to land

File: packages/flutter_tools/lib/src/build_system/targets/dart.dart [hunk 1]
```diff
@@ -55,6 +60,106 @@ List<File> listDartSources(Environment environment) {
...
+    final AssetBundle assetBundle = AssetBundleFactory.instance.createBundle();
+    await assetBundle.build();
+    final Pool pool = Pool(64);
+    await Future.wait<void>(
+      assetBundle.entries.entries.map<Future<void>>((MapEntry<String, DevFSContent> entry) async {
+        final PoolResource resource = await pool.request();
+        try {
+          final File file = fs.file(fs.path.join(environment.outputDir.path, entry.key));
+          file.parent.createSync(recursive: true);
+          await file.writeAsBytes(await entry.value.contentsAsBytes());
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/build_system/targets/dart.dart]: How about testing if `entry.value` is a `DevFSFileContent` and if so use `File.copy()` instead?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_system/targets/dart.dart]: Done

**Comment by jonahwilliams** [comment]
Landed the perquisite PR, will attempt to land if CI passes.

**Review by zanderso — APPROVED** [review]

**Comment by jonahwilliams** [comment]
Per my discussion with @yjbanov, I'm going to ignore web failures on this (since it is unrelated). Will land once build is green

File: dev/devicelab/bin/tasks/gradle_plugin_light_apk_test.dart [hunk 2]
```diff
@@ -230,7 +230,9 @@ Future<void> main() async {
       return TaskResult.success(null);
     } on TaskResult catch (taskResult) {
       return taskResult;
-    } catch (e) {
+    } catch (e, st) {
+      print(e);
+      print(st);
```
Comment (blasten) [inline dev/devicelab/bin/tasks/gradle_plugin_light_apk_test.dart]: nit: this print and the one above

Comment (jonahwilliams) [reply dev/devicelab/bin/tasks/gradle_plugin_light_apk_test.dart]: Oops!

**Comment by jonahwilliams** [comment]
We believe that the web tests are broken and have disabled them. I'm ignoring them for this PR.

**Pull request merged**